src/App.js

```diff
@@ -28,92 +33,106 @@ class App extends Component {
       // * there's nothing left to load
       if (isLoading || !hasMore || error) return;
 
-    const windowHeight = "innerHeight" in window ? window.innerHeight : document.documentElement.offsetHeight;	
-    const body = document.body;	
-    const html = document.documentElement;	
-    const docHeight = Math.max(body.scrollHeight, body.offsetHeight, html.clientHeight,  html.scrollHeight, html.offsetHeight);	
-    const windowBottom = windowHeight + window.pageYOffset
-
-      if ((windowBottom >= docHeight) && (this.state.searchValue === "")) {
+      const windowHeight =
+        "innerHeight" in window
+          ? window.innerHeight
+          : document.documentElement.offsetHeight;
+      const body = document.body;
+      const html = document.documentElement;
+      const docHeight = Math.max(
+        body.scrollHeight,
+        body.offsetHeight,
+        html.clientHeight,
+        html.scrollHeight,
+        html.offsetHeight
+      );
+      const windowBottom = windowHeight + window.pageYOffset;
+
+      if (windowBottom >= docHeight && this.state.searchValue === "") {
         this.performList();
       }
     };
-    this.handleSearchChange = this.handleSearchChange.bind(this);
-    this.performList = this.performList.bind(this);
-    this.performSearch = this.performSearch.bind(this)
-    this.checkKey = this.checkKey.bind(this);
-    this.newSearch = this.newSearch.bind(this);
-    this.tryAgain = this.tryAgain.bind(this);
   }
 
-
   handleSearchChange(event) {
-    this.setState({
-      searchValue: event.target.value
-    });
+    let query = event.target.value.toLowerCase();
+    const filteredMovies = this.state.movies.filter(movie =>
+      movie.Title.toLowerCase().includes(query)
+    );
+
+    if (query.length === 0) {
+      this.performList(false);
```

Comment:
I think setstate listIndex back to 1 if the query has changed to empty so that it starts listing from page 1, else it just lists from the old listIndex